### DISCUSSION TITLE
Not able to generate iri in api-platform

### DISCUSSION THREAD

**Discussion by Muneer-Shafi** [symfony/symfony#57712] [Q&A]
not able to generate iri in api-platform because my ID contains special characters

**Comment by stof** [comment]
why reporting this to Symfony and not to ApiPlatform ?

**Comment by alexandre-le-borgne** [comment]
You may need to look into generating a slug and using it in the iri instead of the id.

You can also add a new unique column which would be your new identifier, without using any special characters (auto increment, uuid, etc) and rename your current id column, no longer using it as an identifier, but leaving a uniqueness constraint.
